sshd/issue_invite.sh

```diff
@@ -10,86 +10,97 @@ set -e
 INVITE_CODE=
 USERNAME=getter
 AUTOMATED=false
+KEY=
 
 function usage () {
-  echo "$0 [-u username] [-i invite code] [-a]"
+  echo "$0 [-u username] [-i invite code] [-a] [-k key]"
   echo "  -i: invite code (if unspecified, a new invite code is generated)"
   echo "  -u: username (default: getter)"
   echo "  -a: do not output complete invite URL"
+  echo "  -k: public key (if unspecified, a new key is generated)"
   echo "  -h, -?: this help message"
   exit 1
 }
 
-while getopts i:u:ah? opt; do
+while getopts i:u:k:ah? opt; do
   case $opt in
     i) INVITE_CODE="$OPTARG" ;;
     u) USERNAME="$OPTARG" ;;
     a) AUTOMATED=true ;;
+    k) KEY="$OPTARG" ;;
     *) usage ;;
   esac
 done
 shift $((OPTIND-1))
 
-TMP=`mktemp -d`
-
-# Either read the supplied invite code or generate a new one.
-if [ -n "$INVITE_CODE" ]
+if [ -n "$KEY" ]
 then
-  INVITE=`echo -n $INVITE_CODE|base64 -d`
+  HOMEDIR=`getent passwd $USERNAME | cut -d: -f6`
+  mkdir -p $HOMEDIR/.ssh
+  echo "$KEY" | base64 -d >> $HOMEDIR/.ssh/authorized_keys
+  echo "" >> $HOMEDIR/.ssh/authorized_keys  # Add newline
+else
```

Comment:
So, the `KEY_OPTS` are extremely important and right now `-k` completely eliminates them. I think you can restructure this so that by the time we get to the `KEY_OPTS` line the public key is a variable or temp file (and reduce a bunch of duplicated lines, e.g. computing the home directory).